driver-core/src/main/com/mongodb/internal/operation/Operations.java

```diff
@@ -504,6 +509,64 @@ MixedBulkWriteOperation bulkWrite(final List<? extends WriteModel<? extends TDoc
                 .let(toBsonDocument(options.getLet()));
     }
 
+    <TResult> CommandReadOperation<TResult> runCommand(final Bson command, final Class<TResult> resultClass) {
+        notNull("command", command);
+        notNull("resultClass", resultClass);
+        return new CommandReadOperation<TResult>(namespace.getDatabaseName(), toBsonDocument(command), codecRegistry.get(resultClass));
+    }
+
+
+    DropDatabaseOperation dropDatabase() {
+        return new DropDatabaseOperation(namespace.getDatabaseName(), getWriteConcern());
+    }
+
+
+    CreateCollectionOperation createCollection(final String collectionName, final CreateCollectionOptions createCollectionOptions,
```

Comment:
It's good to not have this code duplicated anymore.

Note that I have not really checked that the code in `MongoDatabaseImpl` and `MongoOperationPublished` was the same as this new code.